xmtp_mls/src/subscriptions/stream_messages/stream_stats.rs

```diff
@@ -103,9 +101,9 @@ pub struct StreamStats {
 }
 
 impl StreamStats {
-    pub fn new_stats(&self) -> Vec<StreamStat> {
+    pub async fn new_stats(&self) -> Vec<StreamStat> {
         let mut stats = vec![];
-        let mut stats_rx = self.rx.lock();
+        let mut stats_rx = self.rx.lock().await;
```

Comment:
I'm not sure that changing these to async is really necessary. If the lock isn't available immediately then this will likely return nothing as the last reader exhausted the receiver. This could still use a parking_lot Mutex and use try_lock() and if something else is using it return an empty vec.

There are no async operations happening while the lock is held.

Reply:
I'm pulling stats from this channel on a task outside of this fn in another project.